packages/text-input/src/TextInput.tsx

```diff
@@ -171,9 +178,13 @@ const TextInput = React.forwardRef(
     const { usingKeyboard: showFocus } = useUsingKeyboardContext();
     const [hasFocus, setHasFocus] = useState(false);
     const inputSelectorProp = createDataProp('input-selector');
+    const [generatedId, setId] = useState(
+      id || `text-input-${Math.floor(Math.random() * 10000000)}`,
```

Comment:
The bug with what you have now is if someone were to put in `''` or `0` as the prop value, that's falsy, and we'd default to using our generated id.

```suggestion
      id ?? `text-input-${Math.floor(Math.random() * 10000000)}`,
```

Reply:
I think we can simplify this more:

```jsx
const [generatedId] = useState(`text-input-${Math.floor(Math.random() * 10000000)}`);
const id = propsId ?? generatedId;
```

Reply:
Would that simplification account for someone potentially changing the id, or do I still need useEffect to sync?

Reply:
Whenever the id prop changes the component re-runs. The generated id stays the same, but what value we set on the elements changes depending on the presence of the `id` prop. So they are effectively synced 'for free'.